src/NzbDrone.Core/Download/Clients/FreeboxDownload/FreeboxDownloadSettings.cs

```diff
@@ -0,0 +1,87 @@
+using System.Text.RegularExpressions;
+using FluentValidation;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+using NzbDrone.Core.Validation.Paths;
+
+namespace NzbDrone.Core.Download.Clients.FreeboxDownload
+{
+    public class FreeboxDownloadSettingsValidator : AbstractValidator<FreeboxDownloadSettings>
+    {
+        public FreeboxDownloadSettingsValidator()
+        {
+            RuleFor(c => c.Host).ValidHost();
+            RuleFor(c => c.Port).InclusiveBetween(1, 65535);
+            RuleFor(c => c.ApiUrl).NotEmpty()
+                                  .WithMessage("'API URL' must not be empty.");
+            RuleFor(c => c.ApiUrl).ValidUrlBase();
+            RuleFor(c => c.AppId).NotEmpty()
+                                 .WithMessage("'App ID' must not be empty.");
+            RuleFor(c => c.AppToken).NotEmpty()
+                                    .WithMessage("'App Token' must not be empty.");
+            RuleFor(c => c.Category).Matches(@"^\.?[-a-z]*$", RegexOptions.IgnoreCase)
+                                    .WithMessage("Allowed characters a-z and -");
+            RuleFor(c => c.DestinationDirectory).IsValidPath()
+                                                .When(c => c.DestinationDirectory.IsNotNullOrWhiteSpace());
+            RuleFor(c => c.DestinationDirectory).Empty()
+                                    .When(c => c.Category.IsNotNullOrWhiteSpace())
+                                    .WithMessage("Cannot use 'Category' and 'Destination Directory' at the same time.");
+            RuleFor(c => c.Category).Empty()
+                                    .When(c => c.DestinationDirectory.IsNotNullOrWhiteSpace())
+                                    .WithMessage("Cannot use 'Category' and 'Destination Directory' at the same time.");
+        }
+    }
+
+    public class FreeboxDownloadSettings : IProviderConfig
+    {
+        private static readonly FreeboxDownloadSettingsValidator Validator = new FreeboxDownloadSettingsValidator();
+
+        public FreeboxDownloadSettings()
+        {
+            Host = "mafreebox.freebox.fr";
```

Comment:
Yes this is the actual URL when connected to the Freebox network. I assumed it could be the case for the most part. If you are not in the same network you will need to set the public IP in most cases.

Reply:
Ahh, that makes it pretty easy then.